src/transformers/models/layoutlm/configuration_layoutlm.py

```diff
@@ -183,6 +183,17 @@ def generate_dummy_inputs(
             raise ValueError("Cannot generate dummy inputs without PyTorch installed.")
         import torch
 
+        # If dynamic axis (-1) we forward with a fixed dimension of 2 samples to avoid optimizations made by ONNX
+        batch_size = compute_effective_axis_dimension(
+            batch_size, fixed_dimension=OnnxConfig.DEFAULT_FIXED_BATCH, num_token_to_add=0
+        )
+
+        # If dynamic axis (-1) we forward with a fixed dimension of 8 tokens to avoid optimizations made by ONNX
+        token_to_add = tokenizer.num_special_tokens_to_add(is_pair)
+        seq_length = compute_effective_axis_dimension(
+            seq_length, fixed_dimension=OnnxConfig.DEFAULT_FIXED_SEQUENCE, num_token_to_add=token_to_add
+        )
+
         input_dict["bbox"] = torch.tensor(
             [
                 [0] * 4,
```

Comment:
Is this necessary?
I am asking because this and `[self.max_2d_positions] * 4,` below are increasing the sequence length by 2 which makes the foward pass fail.

I tried removing those and the ONNX conversion ends up working fine.

Reply:
The additional two tensors are the bounding boxes for the [CLS] and [SEP] tokens. I think its not necessary if we are getting the sequence length using input_ids.shape as input_ids already considers those tokens.

We will need to add it only if we are relying on compute_effective_axis_dimension to compute sequence length.

Should I go ahead and make the change? @michaelbenayoun

Reply:
Okay I understand, thanks!
Then you can go forward with the changes.

Reply:
Thanks! I made the changes.